java-frontend/src/main/java/org/sonar/java/ast/visitors/CognitiveComplexityVisitor.java

```diff
@@ -64,20 +66,49 @@ private CognitiveComplexityVisitor() {
     complexity = 0;
     nesting = 1;
     ignoreNesting = false;
-    flow = new ArrayList<>();
+    locations = new ArrayList<>();
     ignored = new HashSet<>();
   }
 
   public static JavaFileScannerContext.CognitiveComplexity methodComplexity(@Nullable SemanticModel semanticModel, MethodTree methodTree) {
     if (shouldAnalyzeMethod(semanticModel, methodTree)) {
```

Comment:
I am pretty sure we can get rid of the semanticModel parameters in most of the methods here... this is only used to be null tested... we can maybe get rid of this reference alltogether in that class.